framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -12,9 +12,123 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-//  TODO: extends BaseInteroperabilityStore
-// eslint-disable-next-line  @typescript-eslint/no-extraneous-class
-export class MainchainInteroperabilityStore {
-	// public async forward(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
-	// public async bounce(ccu: CCUpdateParams, ccm: CCMsg): Promise<void> {};
+import { codec } from '@liskhq/lisk-codec';
+import { hash } from '@liskhq/lisk-cryptography';
+import { regularMerkleTree } from '@liskhq/lisk-tree';
+import { BaseInteroperabilityStore } from '../base_interoperability_store';
+import {
+	MODULE_ID_INTEROPERABILITY,
+	STORE_PREFIX_CHANNEL_DATA,
+	STORE_PREFIX_OUTBOX_ROOT,
+} from '../constants';
+import { ccmSchema, channelSchema, outboxRootSchema } from '../schema';
+import { CCMsg, CCUpdateParams, ChannelData, SendInternalContext } from '../types';
+
+export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
+	public async appendToOutboxTree(chainID: Buffer, appendData: Buffer) {
+		const channelSubstore = this.getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHANNEL_DATA);
+		const channel = await channelSubstore.getWithSchema<ChannelData>(chainID, channelSchema);
+		const updatedOutbox = regularMerkleTree.calculateMerkleRoot({
+			value: hash(appendData),
+			appendPath: channel.outbox.appendPath,
+			size: channel.outbox.size,
+		});
+		await channelSubstore.setWithSchema(
+			chainID,
+			{ ...channel, outbox: updatedOutbox },
+			channelSchema,
+		);
+
+		return true;
+	}
+
+	public async addToOutbox(chainID: Buffer, ccm: CCMsg) {
```

Comment:
```
    # Set nonce in CCM only if sending from the chain (not if routing)
    if ccm.sendingChainID == ownChainAccount.ID:
        ccm.nonce = ownChainAccount.nonce
        ownChainAccount.nonce += 1
```
seems to be missing in the logic. any reason for this?

Reply:
Where do you see that logic? That might be old and outdated.

Reply:
ah yes, i was looking at the old version. sorry!